src/lightning/data/streaming/functions.py

```diff
@@ -0,0 +1,188 @@
+        inputs: A sequence of input to be processed by the `fn` function.
+            Each input should contain at least a valid filepath.
+        output_dir: The folder where the processed data should be stored.
+        num_workers: The number of workers to use during processing
+        fast_dev_run: Whether to use process only a sub part of the inputs
+        num_nodes: When doing remote execution, the number of nodes to use.
+        machine: When doing remote execution, the machine to use.
+
+    """
+
+    if len(inputs) == 0:
+        raise ValueError(f"The provided inputs should be non empty. Found {inputs}.")
+
+    if num_nodes is None or int(os.getenv("DATA_OPTIMIZER_NUM_NODES", 0)) > 0:
+        remote_output_dir = _LightningSrcResolver()(output_dir)
+
+        if remote_output_dir is None or "cloudspaces" in remote_output_dir:
+            raise ValueError(
+                f"The provided `output_dir` isn't valid. Found {output_dir}."
+                " HINT: You can either use `/teamspace/s3_connections/...` or `/teamspace/datasets/...`."
+            )
+
+        inputs = inputs() if callable(inputs) else inputs
+        data_processor = DataProcessor(
+            num_workers=num_workers or os.cpu_count(),
+            remote_output_dir=PrettyDirectory(output_dir, remote_output_dir),
+            fast_dev_run=fast_dev_run,
+            version=None,
+            input_dir=_get_input_dir(inputs),
+        )
+        return data_processor.run(LambdaDataTransformRecipe(fn, inputs))
+    return _execute(
+        f"data-prep-map-{datetime.now().strftime('%Y-%m-%d_%H-%M-%S')}",
+        num_nodes,
+        machine,
+    )
+
+
+def chunkify(
+    fn: Callable[[Any], Any],
+    inputs: Sequence[Any],
+    output_dir: str,
+    chunk_size: Optional[int] = None,
+    chunk_bytes: Optional[int] = None,
+    compression: Optional[str] = None,
+    name: Optional[str] = None,
+    num_workers: Optional[int] = None,
+    fast_dev_run: bool = False,
+    num_nodes: Optional[int] = None,
+    machine: Optional[str] = None,
+) -> None:
+    """This function converts a dataset into chunks possibly in a distributed way."""
+
+    if len(inputs) == 0:
+        raise ValueError(f"The provided inputs should be non empty. Found {inputs}.")
+
+    if chunk_size is None and chunk_bytes is None:
+        raise ValueError("Either `chunk_size` or `chunk_bytes` needs to be defined.")
+
+    if num_nodes is None or int(os.getenv("DATA_OPTIMIZER_NUM_NODES", 0)) > 0:
+        remote_output_dir = _LightningSrcResolver()(output_dir)
+
+        if remote_output_dir is None or "cloudspaces" in remote_output_dir:
+            raise ValueError(
+                f"The provided `output_dir` isn't valid. Found {output_dir}."
+                " HINT: You can either use `/teamspace/s3_connections/...` or `/teamspace/datasets/...`."
+            )
+
+        inputs = inputs() if callable(inputs) else inputs
```

Comment:
`inputs` cannot be a callable with the current signature (also this is at odds with the check on `len` above)

Reply:
removed.